python/src/vmaf/__init__.py

```diff
@@ -1,13 +1,44 @@
 import os
+import subprocess
+
+try:
+    from matplotlib import pyplot as plt
+
+except BaseException:
+    # TODO: importing matplotlib fails on OSX with system python, check what can be done there...
+    # Error reported is:
+    #   RuntimeError: Python is not installed as a framework.
+    #   The Mac OS X backend will not be able to function correctly if Python is not installed as a framework.
+    #   See the Python documentation for more information on installing Python as a framework on Mac OS X.
+    #   Please either reinstall Python as a framework, or try one of the other backends.
+    #   If you are using (Ana)Conda please install python.app and replace the use of 'python' with 'pythonw'.
+    #   See 'Working with Matplotlib on OSX' in the Matplotlib FAQ for more information.
+    plt = None
 
-from vmaf.tools.misc import run_process
 
 # Path to folder containing this file
 VMAF_LIB_FOLDER = os.path.dirname(os.path.abspath(__file__))
 
 # Assuming vmaf source checkout, path to top checked out folder
 VMAF_PROJECT = os.path.abspath(os.path.join(VMAF_LIB_FOLDER, '../../..',))
 
+def to_list(value):
+    """
+    TODO python3: map() and filter() return a generator in python3, however vmaf assumes list
```

Comment:
I'm not sure why `map()` was used so often, perhaps to just keep code more compact.
There is a drastic change in py2/py3 for `map()` (as well as stuff like `dict.keys()` etc)

python3 uses generators in many places now (like `map()`, `range()`, `dict.keys()` etc)

Reply:
Personally i prefer map to list comprehension, since it can be more easily converted into parallel map. What's your take on it?

Reply:
Generators can be cheaper to run basically.
What python2 was doing before is the equivalent of this, ie `map()` in python2 is the same as `to_list(map())` here in this case.

Since you're accessing the outcome of `map()` by array-like index, then I guess there is no other way than turning it into something that can be accessed in such a way.

Perf will depend on the size. If `map()` deals with a million items, obviously then expanding that into a list will be costly (must expand full million items...).

With python3, if `map()` yields a million items, you can process them without ever having the whole million in RAM basically. So depends on usage I think. We can't change the way py3 works though, so must deal with the generator :)

Reply:
Google around a bit, and this may be a thing that could be done. Consider this:
```
>>> data = [1,2,3,None]
>>> a = np.array(data)
>>> a
array([1, 2, 3, None], dtype=object)
>>> a[a != np.array(None)]
array([1, 2, 3], dtype=object)
```

Now if we look at `stats.py` for example this function:
```
    @staticmethod
    def nonemean(my_list):
        return np.mean(to_list(filter(lambda x: x is not None, my_list)))
```

Looks like that we could replace this using `np` only like this:

```
    @staticmethod
    def nonemean(my_list):
        a = np.array(my_list)
        return np.mean(a[a != np.array(None)])
```

I'm not super familiar with numpy, and it does look like it hates `None` (whereas pandas seems fine with `None`...).

I found the trick here: https://stackoverflow.com/questions/25254929/efficient-way-of-removing-nones-from-numpy-array

Not sure which is better performance-wise: the np `a[a != ..]`, or the `filter()` approach... one would have to benchmark this to find out.